src/user.ts

```diff
@@ -120,6 +120,32 @@ export class User {
         await this.storage.username;
     }
 
+    /**
+     * Check if a username already exists
+     * @param {string} username - a string param that represents the username
+     * @returns {boolean} true if username already exists and false if doesn't exists
+     */
+    async usernameExists(username: string): Promise<boolean> {
+        const normalizedUsername = username.normalize('NFKC');
+
+        const fullUsername = TonomyUsername.fromUsername(
+            normalizedUsername,
+            AccountType.PERSON,
+            getSettings().accountSuffix
+        );
+
+        try {
+            (await User.getAccountInfo(fullUsername)) as any; // Throws error if username is taken
```

Comment:
you can remove `() as any` now, and the comment `// Throw...` as it's clear in the code